### ISSUE TITLE
Error in version 1.21.0 SNMP

### ISSUE THREAD

**Issue by flopp999** [influxdata/telegraf#10281]
### Relevent telegraf.conf

```
[[inputs.snmp]]
version = 3
auth_protocol = "SHA"
sec_level = "authPriv"
priv_protocol = "AES"
oid = "1.4.6.1.4.1.21796.3.4.3.1.6.2"
```

### System info

Telegraf 1.21.0 Debian 11

### Expected behavior

No errors

### Actual behavior
E! [inputs.snmp] Error in plugin: agent udp://xx.xx.xx.xx:ppp: converting 'ǆ' (OID .1.4.6.1.4.1.21796.3.4.3.1.6.2) for field 1.4.6.1.4.1.21796.3.4.3.1.6.2: invalid conversion type '1.4.6.1.4.1.21796.3.4.3.1.6.2'

After upgrade to 1.21.0 above error appear

**Comment by MyaLongmire** [comment]
Hi @flopp999! is this your full config? Please checkout the `README.md` for the correct way to layout your config

**Comment by reimda** [comment]
@flopp999 could you double check the oid in your config? The first 4 in the oid is almost certainly wrong. I think you want 1.3.6... instead of 1.4.6...

http://oidref.com/1.3.6.1.4.1.21796.3.4.3.1.6

**Comment by flopp999** [comment]
Hi
I changed it to not show the correct value.

This is the OID that I collect
1.3.6.1.4.1.21796.3.3.3.1.6.2

It was working fine until this morning when telegram 1.21 got installed.
Before it was 1.20.4

**Comment by MyaLongmire** [comment]
Is it possible to get more of your config, if you are not comfortable posting it here you can send it to me on slack. To get to the bottom of this I need the correct oid and how you are setting up fields and tables (I do not need you passwords or version 3 security info).

**Comment by flopp999** [comment]
here you go
```
[[inputs.snmp]]
  agents = [ "udp://10x.xx.xx.xx:161" ]
 #   ## Timeout for each SNMP query.
    timeout = "5s"
 #   ## Number of retries to attempt within timeout.
    retries = 1
 #   ## SNMP version, values can be 1, 2, or 3
    version = 3
 #   ## Agent host tag; the tag used to reference the source host
     agent_host_tag = "poseidon"
    ## SNMPv3 auth parameters
    sec_name = "user01"
    auth_protocol = "SHA"
    auth_password = "xxxxxx"
    sec_level = "authPriv"
    priv_protocol = "AES"
    priv_password = "xxxxxxx"

  [[inputs.snmp.field]]
     oid = "1.3.6.1.4.1.21796.3.3.3.1.6.1"

   [[inputs.snmp.field]]
     oid = "1.3.6.1.4.1.21796.3.3.3.1.6.2"

   [[inputs.snmp.field]]
     oid = "1.3.6.1.4.1.21796.3.3.3.1.6.3"

   [[inputs.snmp.field]]
     oid = "1.3.6.1.4.1.21796.3.3.3.1.6.4"

   [[inputs.snmp.field]]
     oid = "1.3.6.1.4.1.21796.3.3.3.1.6.5"
```

**Comment by flopp999** [comment]
I see that lots of people having problem with SNMP.

What is the solution for all this errors, why did you change to GOSMI?

**Comment by MyaLongmire** [comment]
We are trying out best to address all the issues, there are multiple prs up already. We switched to `gosmi` as it is a lot quicker than calling `snmptranslate` and parsing the output. Please be patient as we try to work out all the kinks with the switch.

**Comment by deuntje** [comment]
Please add support for older versions in your repo, in the [Debian repo](https://repos.influxdata.com/debian/pool/stable/t/telegraf/) I can only find the latest broken version, downgrading using apt install telegraf=1.20.4-1 is not possible. For these cases it would be very nice to have the latest X releases available on your repo. I had the luck that I had an old .deb file in my /var/cache/apt/archive directory, otherwise I had some more challenges to downgrade.

**Comment by duc996** [comment]
+1 for adding older version in your repo.

**Comment by MyaLongmire** [comment]
There have been many snmp bugs fixed from 1.20.0 to 1.20.2. Please update to 1.20.2 and if you are still having problems open another issue with your config, mibs, and errors.

Adding support for older versions is a whole different issue. Please make a feature request and someone on the team will get to it as soon as we can. 

Thank you!

**Comment by flopp999** [comment]
> There have been many snmp bugs fixed from 1.20.0 to 1.20.2. Please update to 1.20.2 and if you are still having problems open another issue with your config, mibs, and errors.
> 
> Adding support for older versions is a whole different issue. Please make a feature request and someone on the team will get to it as soon as we can.
> 
> Thank you!

Hi, shall I downgrade?

I have 1.20.4 now

**Comment by flopp999** [comment]
I now found the problem.
I need to have name = "xyz" otherwise it will not work.
